src/compiler/utilities.ts

```diff
@@ -2527,6 +2528,12 @@ export function isPartOfTypeNode(node: Node): boolean {
     return false;
 }
 
+function isPartOfTypeExpressionWithTypeArguments(node: Node) {
+    return isJSDocImplementsTag(node.parent)
+        || isJSDocAugmentsTag(node.parent)

```

Comment:
need a test of `@extends/augments` to go with this check (unless maybe there's already a test like one this PR adds for `@implements`?).

It looks like `extends` forbids expressions with type arguments in TS -- why? Should this apply to JS as well?

Reply:
@sandersn Both `@extends/@augments` and `@implements` use expressions with type arguments (`parseExpressionWithTypeArgumentsForAugments`). That's why I opted to incorporate `isJSDocAugmentsTag` into the condition.

https://github.com/microsoft/TypeScript/blob/5eda842be00e19790ce08d243e0bfbfde6dda4e3/src/compiler/parser.ts#L9432-L9440